src/socket.cc

```diff
@@ -341,44 +341,45 @@ rtp_error_t uvgrtp::socket::__sendtov(
 {
 #ifndef _WIN32
     int sent_bytes = 0;
-    struct mmsghdr *headers = new struct mmsghdr[buffers.size()];
-    struct mmsghdr *hptr = headers;
+    struct msghdr *headers = new struct msghdr[buffers.size()];
+    struct msghdr *hptr = headers;
 
     for (size_t i = 0; i < buffers.size(); ++i) {
-        headers[i].msg_hdr.msg_iov        = new struct iovec[buffers[i].size()];
-        headers[i].msg_hdr.msg_iovlen     = buffers[i].size();
-        headers[i].msg_hdr.msg_name       = (void *)&addr;
-        headers[i].msg_hdr.msg_namelen    = sizeof(addr);
-        headers[i].msg_hdr.msg_control    = 0;
-        headers[i].msg_hdr.msg_controllen = 0;
+        headers[i].msg_iov        = new struct iovec[buffers[i].size()];
+        headers[i].msg_iovlen     = buffers[i].size();
+        headers[i].msg_name       = (void *)&addr;
+        headers[i].msg_namelen    = sizeof(addr);
+        headers[i].msg_control    = 0;
+        headers[i].msg_controllen = 0;
 
         for (size_t k = 0; k < buffers[i].size(); ++k) {
-            headers[i].msg_hdr.msg_iov[k].iov_len   = buffers[i][k].first;
-            headers[i].msg_hdr.msg_iov[k].iov_base  = buffers[i][k].second;
-            sent_bytes                             += buffers[i][k].first;
+            headers[i].msg_iov[k].iov_len   = buffers[i][k].first;
+            headers[i].msg_iov[k].iov_base  = buffers[i][k].second;
+            sent_bytes                     += buffers[i][k].first;
         }
     }
 
     ssize_t npkts = (flags_ & RCE_NO_SYSTEM_CALL_CLUSTERING) ? 1 : 1024;
     ssize_t bptr  = buffers.size();
 
     while (bptr > npkts) {
-        if (sendmmsg(socket_, hptr, npkts, flags) < 0) {
-            log_platform_error("sendmmsg(2) failed");
+        if (sendmsg(socket_, hptr, flags) < 0) {
+            log_platform_error("sendmsg(2) failed");
             return RTP_SEND_ERROR;
```

Comment:
Feel free to disregard this comment but I'm not sure if this is a good idea. `sendmmsg(2)` has been chosen intentionally to decrease latency so removing it entirely for all Unix variants seems quite a sacrifice. Maybe it would be more appropriate to check if the code is running on Mac and use `sendmsg(2)` in that case instead of removing all uses of `sendmmsg(2)`

Reply:
I'd like to at least do some testing before making changes. There is still some frame loss in uvgRTP with high frame rates and one of the potential solutions would be to slow down or pace the sending somehow and this probably works better with sendmsg(2).

Reply:
I'll try to implement both, maybe with a compile time option, so they can be compared.

Reply:
You can, but I can also just compare it with master also.